Tests/UnitTests/EdgeNetworkServiceTests.swift

```diff
@@ -338,12 +419,13 @@ class EdgeNetworkServiceTests: XCTestCase {
 
         // test
         let mockHttpConnection = HttpConnection(data: stringResponseBody.data(using: .utf8),
-                                                response: HTTPURLResponse(url: url, statusCode: 503, httpVersion: nil, headerFields: nil),
+                                                response: HTTPURLResponse(url: url, statusCode: 503, httpVersion: nil, headerFields: ["Retry-After": "60"]),
```

Comment:
We should only set retryInterval when no call to onComplete and should retry.

Reply:
Looks like the test name may be wrong, but the error code of 503 is recoverable therefore we should retry this hit, and subsequently set the retry interval. Is this correct?

Reply:
Went ahead and updated this test name.